lib/lib_rss.php

```diff
@@ -444,8 +444,14 @@ function stripHtmlMetaCharset(string $html): string {
 function enforceHttpEncoding(string $html, string $contentType = ''): string {
 	$httpCharset = preg_match('/\bcharset=([0-9a-z_-]{2,12})$/i', $contentType, $matches) === 1 ? $matches[1] : '';
 	if ($httpCharset == '') {
-		// No charset defined by HTTP, do nothing
-		return $html;
+		// No charset defined by HTTP
+		if (preg_match('/<meta\s[^>]*charset\s*=[\s\'"]*UTF-?8\b/i', substr($html, 0, 2048))) {
+			// Favour UTF-8 even if defined too deep in HTML for DOMDocument
```

Comment:
Maybe something more like this? "Favor" strikes me as misleading. It's just that it's a better default in 2024 than iso-8859-1 or whatever the PHP default is.
```suggestion
			// Default to UTF-8 if defined too deep in HTML for DOMDocument
```

Reply:
What I meant is that we do some additional efforts for UTF-8, by searching some more in the HTML content

Reply:
PHP has no default for that, but HTTP has: Windows-1252 (similar to ISO-8859-1)

Reply:
New rewording suggestion

Reply:
Oh wait, I read the code here completely in reverse because it doesn't relate to the OP at all afaict.

OP: "15k whitespace then 1.2k of scripts "

This code: 2048 characters (or probably even just bytes)

Reply:
This PR also trims the whitespace first